commit 34d301cc161e06ef612914a9e35ce2878417cbe1
Author: Brian Russell <brussell@brocade.com>
Date:   Fri Feb 24 17:47:11 2017 +0000

    vxlan: don't allow overwrite of config src addr
    
    [ Upstream commit 1158632b5a2dcce0786c1b1b99654e81cc867981 ]
    
    When using IPv6 transport and a default dst, a pointer to the configured
    source address is passed into the route lookup. If no source address is
    configured, then the value is overwritten.
    
    IPv6 route lookup ignores egress ifindex match if the source address is set,
    so if egress ifindex match is desired, the source address must be passed
    as any. The overwrite breaks this for subsequent lookups.
    
    Avoid this by copying the configured address to an existing stack variable
    and pass a pointer to that instead.
    
    Fixes: 272d96a5ab10 ("net: vxlan: lwt: Use source ip address during route lookup.")
    
    Signed-off-by: Brian Russell <brussell@brocade.com>
    Acked-by: Jiri Benc <jbenc@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1376_34d301cc161e06ef612914a9e35ce2878417cbe1.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1376_34d301cc161e06ef612914a9e35ce2878417cbe1.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
